### PR TITLE
Update D-Bus documentation

### PR DESCRIPTION
**Pull request by imobachgs**
- Update network D-Bus docs
- Fix the generation of the Users1 iface

### CODE REVIEW

File: doc/dbus/org.opensuse.Agama1.Network.Connection.IP.doc.xml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    <!--
+     Connection ID.
+
+     Unique identifier of the network connection. It may or not be the same that the used by the
+     backend. For instance, when using NetworkManager (which is the only supported backend by
+     now), it uses the original ID but appending a number in case the ID is duplicated.
+     -->
+    <property name="Id" type="s" access="read"/>
+    <property name="Interface" type="s" access="readwrite"/>
+  </interface>
```
Comment (mvidner) [inline doc/dbus/org.opensuse.Agama1.Network.Connection.IP.doc.xml:49]: All the above interfaces should be omitted. Each `foo.doc.xml` file should only contain one `<interface>` element, `<interface name="foo">`

That is why https://github.com/openSUSE/agama/actions/runs/6300720786/job/17107419428?pr=769 is failing

Comment (imobachgs) [reply doc/dbus/org.opensuse.Agama1.Network.Connection.IP.doc.xml]: Thanks!. Now I got it. I have fixed those issues.

File: doc/dbus/org.opensuse.Agama1.Network.Connection.Match.doc.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <!--
+     List of driver
+     -->
+    <property name="Interface" type="as" access="readwrite"/>
+    <!--
+     List of kernel options
+     -->
+    <property name="Kernel" type="as" access="readwrite"/>
+    <!--
+     List of paths
```
Comment (mvidner) [inline doc/dbus/org.opensuse.Agama1.Network.Connection.Match.doc.xml:17]: What paths are these? Object paths? Filesystem paths? Of what?

File: doc/dbus/org.opensuse.Agama1.Network.Connections.doc.xml [hunk 3]
```diff
@@ -41,13 +35,15 @@
...
-     Apply:
      Applies the network configuration.
 
      It includes adding, updating and removing connections as needed.
      -->
     <method name="Apply">
     </method>
+    <signal name="ConnectionAdded">
+      <arg name="id" type="s"/>
+      <arg name="path" type="s"/>
```
Comment (mvidner) [inline doc/dbus/org.opensuse.Agama1.Network.Connections.doc.xml:46]: The same *path* kind as above I guess, what is it? Should it be `type="o"`?

File: doc/dbus/org.opensuse.Agama1.Network.Connection.Match.doc.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<node>
+  <interface name="org.opensuse.Agama1.Network.Connection.Match">
+    <!--
+     List of driver
+     -->
+    <property name="Driver" type="as" access="readwrite"/>
+    <!--
+     List of driver
```
Comment (mvidner) [inline doc/dbus/org.opensuse.Agama1.Network.Connection.Match.doc.xml:9]:
```suggestion
     List of interface
```

**Review by mvidner — COMMENTED** [review]
Yay the check passes now. Just some questions about the API description

File: doc/dbus/org.opensuse.Agama1.Network.Connections.doc.xml (hunk 3 above)
Comment (imobachgs) [reply doc/dbus/org.opensuse.Agama1.Network.Connections.doc.xml:46]: No, this one is a D-Bus path, not related to the kernel path of the device. But, yes, it should be an `o`.

**Review by mvidner — APPROVED** [review]
Thanks!

**Pull request merged**